src/frontends/pytorch/src/transforms/aten_index_replacer.cpp

```diff
@@ -0,0 +1,273 @@
+
+                if (const_input != nullptr) {
+                    const auto& attrs = const_input->get_attrs();
+                    if (attrs.find("none_value") != attrs.end()) {
+                        masked_indicies.push_back(ids[i]);
+                        is_masked_bool.push_back(false);
+                        continue;
+                    }
+                }
+                auto id_dtype = ids[i].get_node_shared_ptr()->get_element_type();
+                if (id_dtype == element::boolean || id_dtype == element::u8) {
+                    auto idx = std::make_shared<ov::op::v0::Convert>(ids[i], element::u8);
+                    auto nonzero = std::make_shared<ov::op::v3::NonZero>(idx);
+                    auto input_order = v0::Constant::create(element::i64, Shape{2}, {1, 0});
+                    auto masked_id = std::make_shared<v1::Transpose>(nonzero, input_order);
+                    masked_indicies.push_back(masked_id);
+                    is_masked_bool.push_back(true);
+                } else {
+                    masked_indicies.push_back(ids[i]);
+                    is_masked_bool.push_back(false);
+                }
+                advanced_ids.push_back(i);
+            }
+
+            // all indicies prim::Constant(None), return input as is
+            if (advanced_ids.size() == 0) {
+                copy_runtime_info({index_op, input_node}, input_node);
+                replace_node(index_op, input_node);
+                return true;
+            }
+            // perform gather for single element case
+            if (advanced_ids.size() == 1) {
+                auto index = masked_indicies[advanced_ids[0]];
+                index = std::make_shared<v0::Convert>(index, element::i64);
+                auto dim = v0::Constant::create(element::i64, Shape{}, {advanced_ids[0]});
+                if (is_masked_bool[advanced_ids[0]]) {
+                    auto gather = std::make_shared<v8::GatherND>(input_node, index);
+                    copy_runtime_info({index_op, input_node, indicies}, gather);
+                    replace_node(index_op, gather);
+                    return true;
+                }
+                auto gather = std::make_shared<v8::Gather>(input_node, index, dim);
+                copy_runtime_info({index_op, input_node, indicies}, gather);
+                replace_node(index_op, gather);
+                return true;
+            }
+            auto adv_idx_count = advanced_ids.size();
+            auto rank = input_node->get_input_partial_shape(0).rank();
+            if (rank.is_dynamic()) {
+                FRONT_END_CHECK_IMPLEMENTED(false, "indexing for tensor with dynamic rank is not implemented ");
+            }
+            auto input_shape = std::make_shared<v3::ShapeOf>(input_node);
+            auto zero = v0::Constant::create(element::i64, Shape{}, {0});
+            auto input_dims = std::make_shared<v1::Split>(input_shape, zero, rank.get_length());
```

Comment:
While testing it on Cityscapes/mask_rcnn_R_50_FPN models we saw that there might be an issue with Split op when splitting dimension of 3 into 2 splits due to OV limitations.

Reply:
ov `Split` only works for the cases when dimensions are divisible, like in this case where we split shape by rank